src/games/strategy/engine/framework/ui/NewGameChooser.java

```diff
@@ -246,26 +246,26 @@ private synchronized static void synchronizedClear() {
   private void refreshGameList() {
     m_gameList.setEnabled(false);
     final NewGameChooserEntry selected = getSelected();
-    SwingUtilities.invokeLater(new Runnable() {
-      @Override
-      public void run() {
         try {
           refreshNewGameChooserModel();
           m_gameListModel = getNewGameChooserModel();
           m_gameList.setModel(m_gameListModel);
-          if (selected != null) {
-            final String name = selected.getGameData().getGameName();
-            final NewGameChooserEntry found = m_gameListModel.findByName(name);
-            if (name != null) {
-              m_gameList.setSelectedValue(found, true);
-            }
+          if(selected != null){
+            selectGame(selected.getGameData().getGameName());
           }
         } finally {
           m_gameList.setEnabled(true);
         }
+  }
+  
+  private void refreshGameListBackground(){
```

Comment:
This is not actually running the 'refreshGameList' in background. SwingUtlitilies.invokeLater will put it at the end of the AWT event thread queue. Anything running on that thread blocks user input, so we're effectively queue the 'refreshGameList' for execution in the foreground.

I notice this while testing, when I click 'choose game', I'm see what looks to be the map parsing delay. So it seems like we are refreshing the game list in the foreground when it is needed. IMO this is not such a bad thing, but it does mean the cache code we have is no longer being used nor  would be needed.